dev/release/verify-release-candidate.sh

```diff
@@ -895,6 +895,8 @@ ensure_source_directory() {
     fi
   fi
 
+  # Ensure ARROW_SOURCE_DIR is a safe.directory for git
+  git config --global --add safe.directory $ARROW_SOURCE_DIR
```

Comment:
Hmm, this script can be run by developers on their local machine, can we run the `git config` only on CI configs? (for example in the parent script or yaml file)

Reply:
We can add it to the docker-compose.yml:
```
  conda-verify-rc:
    image: ubuntu:${UBUNTU}
    volumes:
      - .:/arrow:delegated
      - ${DOCKER_VOLUME_PREFIX}conda-ccache:/ccache:delegated
    shm_size: '1gb'
    environment:
      <<: *ccache
      CMAKE_GENERATOR: Ninja
      DEBIAN_FRONTEND: "noninteractive"
      DOTNET_SYSTEM_GLOBALIZATION_INVARIANT: 1
      TEST_APT: 0  # would require docker-in-docker
      TEST_YUM: 0
      USE_CONDA: 1
    command: >
      /bin/bash -c "
        apt update -y && apt install -y curl git gnupg tzdata wget &&
        git config --global --add safe.directory /arrow &&
        /arrow/dev/release/verify-release-candidate.sh $${VERIFY_VERSION} $${VERIFY_RC}"
```
If we want to add it on the CI files only (not even `docker-compose.yml` we can modify the archery call `arrow/dev/tasks/verify-rc/github.linux.amd64.docker.yml` and add an environment variable to manage it:
```
      - name: Execute Docker Build
        shell: bash
        run: |
          archery docker run \
            -e VERIFY_VERSION="{{ release|default("") }}" \
            -e VERIFY_RC="{{ rc|default("") }}" \
            -e TEST_DEFAULT=0 \
            -e TEST_{{ target|upper }}=1 \
            -e SET_GIT_SAFE_DIR=1 \
            {{ distro }}-verify-rc
```
Would you be ok with modifying the docker-compose as shown above?